std/typecons.d

```diff
@@ -7143,3 +7143,265 @@ unittest
         ubyte, ubyte, T3, T3,
     );
 }
+
+/**
+ * Get the value of a field or property member. The member's name
+ * is resolved at runtime, rather than at compile-time.
+ *
+ * Note that currently only `pure @safe` property methods are supported.
```

Comment:
@klickverbot The purpose of these functions is to select which property to access out of some set *at runtime*. If even **one** property in the set is not `@safe`, then the dispatch function can't be `@safe` either.

This would be a pretty severe restriction, when you consider that the set of accessible properties is generated automatically at compile time, based mostly on the return type of the property. Thus, non-`pure @safe` functions can easily end up in the set, *even if the user has no intention of ever accessing them*.

As a concrete example, we discovered this problem because without the `pure @safe` filter, `getField!int()` cannot be used on `class` objects from `pure` code on 32-bit, because `Object.toHash()` has a compatible return type, but is not `pure`.